sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -186,8 +186,13 @@ def __init__(self, packages, options, environment_version, pipeline_url):
     if job_type.startswith('FNAPI_'):
       self.debug_options = self.debug_options or DebugOptions()
       self.debug_options.experiments = self.debug_options.experiments or []
-      if not self.debug_options.lookup_experiment(
-          'runner_harness_container_image'):
+      if self.debug_options.lookup_experiment(
+          'runner_harness_container_image') or _use_unified_worker(
+              self.debug_options):
+        # Don't add the default image overwrite if user overwrites or
```

Comment:
This comment repeats the logic; I thought you might want to explain why we don't want to pin the image for UW. 
Consider: "In case of using UW, harness container should be defined by DF service unless explicitly overridden."

Reply:
Updates the comment.